### PR TITLE
[dotnet] Make sure all references to Pulumi types now are suffixed with the global modifier

### LINKED ISSUE #10684: Support nested namespaces in C# codegen
When building a community package or provider, it's possible you might want to use a distinct namespace for the Pulumi package that's created.

You can specify a `RootNamespace` like so:

```
		CSharp: &tfbridge.CSharpInfo{
			RootNamespace: "Lbrlabs",
			PackageReferences: map[string]string{
				"Pulumi": "3.*",
			},
		},
```

Which generates a nuget package like `Lbrlabs.Something`

However, you probably want it to be clear it's a Pulumi package, so ideally you'd be able to do:

```
		CSharp: &tfbridge.CSharpInfo{
			RootNamespace: "Lbrlabs.Pulumi",
			PackageReferences: map[string]string{
				"Pulumi": "3.*",
			},
		},
```

If you try this, the codegen fails:

```
  Determining projects to restore...
  Restored /Users/lbriggs/src/github/lbrlabs/pulumi-grafana/sdk/dotnet/Lbrlabs.Pulumi.Grafana.csproj (in 157 ms).
/Users/lbriggs/src/github/lbrlabs/pulumi-grafana/sdk/dotnet/Config/Config.cs(33,40): error CS0234: The type or namespace name 'Config' does not exist in the namespace 'Lbrlabs.Pulumi' (are you missing an assembly reference?) [/Users/lbriggs/src/github/lbrlabs/pulumi-grafana/sdk/dotnet/Lbrlabs.Pulumi.Grafana.csproj]

Build FAILED.

/Users/lbriggs/src/github/lbrlabs/pulumi-grafana/sdk/dotnet/Config/Config.cs(33,40): error CS0234: The type or namespace name 'Config' does not exist in the namespace 'Lbrlabs.Pulumi' (are you missing an assembly reference?) [/Users/lbriggs/src/github/lbrlabs/pulumi-grafana/sdk/dotnet/Lbrlabs.Pulumi.Grafana.csproj]
    0 Warning(s)
    1 Error(s)
```

The errored line is:

```
namespace Lbrlabs.Pulumi.Grafana
// some other stuff
private static readonly Pulumi.Config __config = new Pulumi.Config("grafana");
```

### PR DESCRIPTION
**Pull request by Zaid-Ajaj**
# Description

Fixes #10684 (issue)

To test this, I've updated the `provider-config-schema` test schema and added the root namespace option to the csharp codegen results. I've also added a function example just to make sure that even calls to underlying `Pulumi.Deployment` are suffixed with `::global`

## Checklist

- [x] I have added tests that prove my fix is effective or that my feature works

- [x] I have updated the [CHANGELOG-PENDING](https://github.com/pulumi/pulumi/blob/master/CHANGELOG_PENDING.md) file with my change

- [ ] Yes, there are changes in this PR that warrants bumping the Pulumi Service API version

### CODE REVIEW

File: pkg/codegen/dotnet/gen.go [hunk 1]
```diff
@@ -1380,7 +1380,7 @@ func (mod *modContext) genFunction(w io.Writer, fun *schema.Function) error {
 	// Emit the datasource method.
 	fmt.Fprintf(w, "        public static Task%s InvokeAsync(%sInvokeOptions? options = null)\n",
```
Comment (justinvp) [inline pkg/codegen/dotnet/gen.go:1381]: Should we be fully qualifying other types like `InvokeOptions` as `global::Pulumi.InvokeOptions`? (There are probably other similar cases).

Comment (justinvp) [reply pkg/codegen/dotnet/gen.go:1381]: I almost wonder if we should be fully qualifying _everything_, even types from the `System` namespace. It'll look ugly, but reduce the chances of conflicts. Maybe we don't need to go that far just yet.

Comment (Zaid-Ajaj) [reply pkg/codegen/dotnet/gen.go:1381]: > Maybe we don't need to go that far just yet.

I think the problem arises when when we a fully qualified type such as `Pulumi.Config` that is not marked with `global::`. Other Pulumi types we use are not fully qualified: `Input<T>`, `Output<T>`, `InvokeOptions` and these resolve correctly. I think it is unlikely that we get a type collision with these from the generated types of the schema

**Review by iwahbe — APPROVED** [review]
I have no idea why you have python tests failing. I think you might need to pull `master`. Otherwise LGTM.

**Comment by Zaid-Ajaj** [comment]
@iwahbe tests failed because I didn't commit all codegen changes for other languages (only did `PULUMI_ACCEPT=1 make test_pkg_dotnet`) but now I've done that, let's see if CI is happy now 😄 

The `Update Changelog` check however is still blocking a green check, should that be removed that we now use separate files for change logs? cc @AaronFriel

**Comment by iwahbe** [comment]
bors retry

**Review by Frassle — APPROVED** [review]

**Comment by iwahbe** [comment]
`no-changelog-required` tag to fool the CI system.

**Comment by iwahbe** [comment]
@Zaid-Ajaj It looks like the docs need to be regenerated.

**Comment by Zaid-Ajaj** [comment]
bors retry

**Pull request merged**